### PR TITLE
 logreader/logproto: remove preemption

### PR DESCRIPTION
**Pull request by MrAnno**
The "preemption" logic with NC_FILE_SKIP was an implementation detail of the old wildcard file source.
This is dead code now.

This PR removes code from a "hot path", but I guess this is not a significant performance improvement. :)

### CODE REVIEW

**Comment by MrAnno** [comment]
This PR removes code from a "hot path", but I guess this is not a significant performance improvement. :)

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by MrAnno** [comment]
Questions to reviewers:
- Why could I change the value of `NC_REOPEN_REQUIRED`?
- Can I do the same with `LR_THREADED` (consider persist)?

File: lib/logreader.c [hunk 1]
```diff
@@ -329,9 +328,6 @@ log_reader_fetch_log(LogReader *self)
   gboolean may_read = TRUE;
```
Comment (Kokan) [inline lib/logreader.c]: Would it be possible to also remove `may_read` ?

Comment (MrAnno) [reply lib/logreader.c]: It is a requirement of "multi-read" which is a used feature in `affile/linux-kmsg.c`.

Comment (Kokan) [reply lib/logreader.c]: Yeah, you are right.

**Comment by MrAnno** [comment]
Oops, I forgot to make this WIP. I'll fix the value of LR_THREADED in my next PR.

**Comment by Kokan** [comment]
A merged PR is a good PR :)

**Comment by MrAnno** [comment]
😄

**Pull request merged**